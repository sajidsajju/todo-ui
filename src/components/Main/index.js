import React, { useEffect, useState } from "react";
import { makeStyles } from "@material-ui/core";
import TodoList from "../TodoList";
import TodoForm from "../TodoForm";

const useStyles = makeStyles({
  h1: {
    textAlign: "center",
    fontSize: "4em",
    letterSpacing: "7px",
    color: "#4d4d4d",
  },
});

function Main() {
  const classes = useStyles();
  const initialState = JSON.parse(localStorage.getItem("todos")) || [];

  const [todos, setTodos] = useState(initialState);

  const addTodo = todo => {
    const newTodos = [todo, ...todos];
    setTodos(newTodos);
  };

  const removeTodo = id => {
    const deleteTodo = [...todos].filter(todo => todo.id !== id);

    setTodos(deleteTodo);
  };

  const editTodo = Todo => {
    const newTodos = [...todos].map(todo => {
      if (todo.id === Todo.id) {
        todo.text = Todo.text;
        todo.completed = Todo.completed;
      }
      return todo;
    });
    setTodos(newTodos);
  };

  useEffect(() => {
    localStorage.setItem("todos", JSON.stringify(todos));
  }, [todos]);

  return (
    <>
      <h1 className={classes.h1}>TODO</h1>
      <TodoForm addTodo={addTodo} />
      <TodoList todos={todos} removeTodo={removeTodo} editTodo={editTodo} />
    </>
  );
}

export default Main;
