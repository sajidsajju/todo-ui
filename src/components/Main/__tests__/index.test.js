import React from "react";
import { render, screen } from "@testing-library/react";

import Main from "../";

describe("<Main  />", () => {
  it("should match the snapshot", () => {
    const rendered = render(<Main />);

    expect(rendered.container.firstChild).toMatchSnapshot();
  });

  it("should match the title", () => {
    render(<Main />);

    expect(screen.getByText("TODO")).toBeInTheDocument();
  });
});
