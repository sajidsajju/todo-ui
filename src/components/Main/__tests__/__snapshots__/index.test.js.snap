// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main  /> should match the snapshot 1`] = `
<h1
  class="makeStyles-h1-1"
>
  TODO
</h1>
`;
